File: project-overdue/overdue-parent/overdue-role/src/main/java/cn/sjxy/overdue/service/CourseService.java
package cn.sjxy.overdue.service;
import java.util.List;

import cn.sjxy.overdue.pojo.Course;
import cn.sjxy.overdue.pojo.entity.PageResult;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CourseService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Course> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Course course);
	
	
	/**
	 * 修改
	 */
	public void update(Course course);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Course findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Course course, int pageNum, int pageSize);
	
}

File: project-overdue/overdue-parent/overdue-role/src/main/webapp/js/service/courseService.js
//服务层
app.service('courseService', function ($http) {
    //监管学生
    this.findStuProcess = function () {
        return $http.get("/secretary/findStuProcess");
    }
    //分配任务
    this.assignTask = function (teacherId, couId, studentId) {
        return $http.get("/secretary/assignTask?teacherId=" + teacherId + "&couId=" + couId + "&studentId=" + studentId);
    }
    //查找所有学生
    this.findStudent = function (courseId) {
        return $http.get("/secretary/findStudent?courseId=" + courseId);
    };
    this.judgePro = function (courseId, proId, teaId) {
        return $http.get("/secretary/judgePro?courseId=" + courseId + "&proId=" + proId + "&teaId=" + teaId);
    }
    this.findAllProf = function (courseId) {
        return $http.get("/secretary/findAllProf?courseId=" + courseId);
    }
    //根据课程id查询所有任课教师
    this.showTeacher = function (cId) {
        return $http.get("/secretary/showTeacher?cId=" + cId);
    }
    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../course/findAll.do');
    }
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../course/findPage.do?page=' + page + '&rows=' + rows);
    }
    //查询实体
    this.findOne = function (id) {
        return $http.get('../course/findOne.do?id=' + id);
    }
    //增加
    this.add = function (entity) {
        return $http.post('../course/add.do', entity);
    }
    //修改
    this.update = function (entity) {
        return $http.post('../course/update.do', entity);
    }
    //删除
    this.dele = function (ids) {
        return $http.get('../course/delete.do?ids=' + ids);
    }
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../course/search.do?page=' + page + "&rows=" + rows, searchEntity);
    }
});
